core/api.go

```diff
@@ -304,10 +304,29 @@ type LoadBalancer interface {
 	Next() Member
 }
 
-// HazelcastJSON is a wrapper for json byte array.
+// HazelcastJSONValue is a wrapper for json byte array.
 // JsonString should be a valid json string.
-type HazelcastJSON struct {
-	JSONString []byte
+type HazelcastJSONValue struct {
+	jsonString []byte
+}
+
+func CreateHazelcastJSONValueFromString(jsonString string) *HazelcastJSONValue {
+	return &HazelcastJSONValue{[]byte(jsonString)}
+}
+func CreateHazelcastJSONValue(object interface{}) *HazelcastJSONValue {
+	if object == nil {
+		return nil
+	}
+	byteArray, _ := json.Marshal(object)
+	return &HazelcastJSONValue{byteArray}
+}
+
+func (h *HazelcastJSONValue) Unmarshal(v interface{}) {
+	json.Unmarshal(h.jsonString, &v)
```

Comment:
This method also returns an error, why dont we return it to the user?

Reply:
An assumption here was that server will return valid JSON always. When I reconsider, there is nothing preventing the user from putting invalid JSON, therefore it makes more sense to return an error from Unmarshal. @asimarslan  wdyt ?

Reply:
yes it should just return `json.Unmarshal` all of the results. no need to hide it.